File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
  const playerIndices = {};

  for (let i = 0; i < players.length; i++) {
    playerIndices[players[i]] = i;
  }

  for (let i = 0; i < callings.length; i++) {
    const calledPlayer = callings[i];
    const calledPlayerIndex = playerIndices[calledPlayer];

    if (calledPlayerIndex > 0) {
      const prevPlayer = players[calledPlayerIndex - 1];
      players[calledPlayerIndex - 1] = calledPlayer;
      players[calledPlayerIndex] = prevPlayer;

      playerIndices[calledPlayer] = calledPlayerIndex - 1;
      playerIndices[prevPlayer] = calledPlayerIndex;
    }
  }
  return players;
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    playerIndicies = {}
    
    for i in range(len(players)):
        playerIndicies[players[i]] = i
    
    for i in range(len(callings)):
        calledPlayer = callings[i]
        calledPlayerIdx = playerIndicies[calledPlayer]
        
        if (calledPlayerIdx > 0):            
            tmp = players[calledPlayerIdx - 1]
            players[calledPlayerIdx - 1] = calledPlayer
            players[calledPlayerIdx] = tmp
            
            playerIndicies[calledPlayer] = calledPlayerIdx - 1
            playerIndicies[tmp] = calledPlayerIdx
        
    return players